sys/src/9/amd64/entry.S

```diff
@@ -275,6 +275,77 @@ _warp64:
 	addl	$PGLSZ(1), %edx
 	movl	%edx, PDO(KZERO+7*PGLSZ(1))(%eax)	/* PDE for KZERO 14-16MiB */
 
+	// and up to 64! I tried a loop but it's harder than I thought.
+	// better code welcome.
+	// Warning: you've only got 256M to work with. This is a foundational mistake,
+	// and it's way harder to fix than I thought. So, ugly.
+	// 16 -> 32
+	movl	%edx, PDO(KZERO+8*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+9*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+10*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+11*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+12*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+13*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+14*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+15*PGLSZ(1))(%eax)
+
+	// 32 -> 48
+	movl	%edx, PDO(KZERO+16*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+17*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+18*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+19*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+20*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+21*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+22*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+23*PGLSZ(1))(%eax)
+
+	// 48 -> 64
+	movl	%edx, PDO(KZERO+24*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+25*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+26*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+27*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+28*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+29*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+30*PGLSZ(1))(%eax)
+	addl	$PGLSZ(1), %edx
+	movl	%edx, PDO(KZERO+31*PGLSZ(1))(%eax)
+
+	movl	%edx, PDO(KZERO+16*PGLSZ(1))(%eax)
```

Comment:
Shouldn't this be +32?  Or am I missing something subtle?

Reply:
did I screw this up again? I'll look

Reply:
I had one job ... 
check again ....
it does boot. I expect this is not even needed, I am realizing it may only matter until mmuinit(), but I'll test that later.